codec-stomp/src/main/java/io/netty/handler/codec/stomp/DefaultStompFrame.java

```diff
@@ -60,7 +66,7 @@ public StompFrame retainedDuplicate() {
 
     @Override
     public StompFrame replace(ByteBuf content) {
-        return new DefaultStompFrame(command, content);
+        return new DefaultStompFrame(command, content, new DefaultStompHeaders().set(headers));
```

Comment:
Please use the same strategy as https://github.com/netty/netty/commit/942b993f2b9781ff2126ff92a6be5b975dfc72ed#diff-17c17dd59c9912dd08e4dc95fa04bcab. In summary adding a `copy()` method on `DefaultStompHeaders` which leverages `DefaultHeaders#copy()`.

Reply:
I looked to this strategy but `DefaultStompHeaders extends DefaultHeader implements StompHeaders` we can't add `StompHeaders#copy()` method it will be clashes. Maybe we can add method with name like a `StompHeaders#deepCopy()` or another way ?

Reply:
We can't modify `StompHeaders` because it is an interface and that would be an API breaking change. However we can modify `DefaultStompHeaders` and do the following:

```java
    @Override
    public DefaultStompHeaders copy() {
        DefaultStompHeaders copy = new DefaultStompHeaders();
        copy.addImpl(this);
        return copy;
    }
```

We could also modify `DefaultHeaders` so we can consolidate the `copy.addImpl`, and any other work which maybe necessary in the future, but this maybe unnecessary.

Reply:
So we can, but in `DefaultStompHeadersSubframe` used interface `StompHeaders`, how we can used implementation method maybe I misunderstood you.

Reply:
@Scottmitch I understand that you are busy, but I would like to quickly close this defect.

Reply:
> how we can access to method that not present in interface without casting or other trick? 

Unfortunately before jdk8 (which provides the `default` key word) extending interfaces is not possible without breaking API changes. So we have a few options:

1. Add a `DefaultStompHeaders copy()` method (as suggested above)
- change the member variable in `DefaultStompHeadersSubframe` so it is declared as type `DefaultStompHeaders`.
- make the new constructor in `DefaultStompHeadersSubframe` which currently takes a `StompHeaders` instead take a `DefaultStompHeaders` and make it package private
2. "break" the StompHeaders API and add a `StompHeaders copy()` method (and also override this method in `DefaultStompHeaders` as suggested above).

@normanmaurer - my guess is you would prefer the "non-breaking" API change?